### PR TITLE
sql: added support for CREATE VIEW IF NOT EXISTS

### PR DESCRIPTION
**Pull request by kujon**
Addresses #1899.

### CODE REVIEW

**Comment by wangandi** [comment]
Thank you for your contribution, @kujon! That was very fast!

It looks like some of the unit tests are not compiling because of the change to the Plan enum definition. It should be a quick fix. Just add the `if_not_exists`. You can run `cargo test` to ensure that the unit tests compile and pass.  

Regarding adding additional tests, you should add a unit test in `src/sql-parser/tests/sqlparser_common.rs` to test that the new syntax parses correctly. 

As for system tests, I would say:
* add a test for the case where the view does not already exist in test/testdrive/createdrop.td
* add a test for the case where the view does already exist in test/testdrive/dependencies.td . Then verify that the `CREATE IF NOT EXISTS` command did not cause the existing view to be replaced. 

For more details on testing, check out [doc/developer/testing.md](https://github.com/MaterializeInc/materialize/blob/master/doc/developer/testing.md).

File: src/pgwire/protocol.rs [hunk 1]
```diff
@@ -693,7 +693,7 @@ where
             ExecuteResponse::CreatedIndex { existed } => created!(existed, "42710", "index"),
             ExecuteResponse::CreatedSource { existed } => created!(existed, "42710", "source"),
             ExecuteResponse::CreatedSink { existed } => created!(existed, "42710", "sink"),
-            ExecuteResponse::CreatedView => command_complete!("CREATE VIEW"),
+            ExecuteResponse::CreatedView { existed } => created!(existed, "42710", "view"),
```
Comment (kujon) [inline src/pgwire/protocol.rs]: Is this errcode correct?

Comment (wangandi) [reply src/pgwire/protocol.rs]: Yup. 42710 stands for "duplicate object". https://www.postgresql.org/docs/12/errcodes-appendix.html

**Comment by kujon** [comment]
Hi @wangandi,

I've addressed your comments and I believe we're ready for the code review!

**Comment by kujon** [comment]
All suggestions done @wangandi

**Comment by kujon** [comment]
cheers!

**Pull request merged**